crates/ursa-tracker/src/main.rs

```diff
@@ -0,0 +1,222 @@
+use crate::{
+    ip_api::get_ip_info,
+    types::{Node, TrackerRegistration, PrometheusDiscoveryChunk},
+};
+use axum::http::StatusCode;
+use axum::{
+    extract::{ConnectInfo, Extension},
+    routing::{get, post},
+    Json, Router,
+};
+use hyper::HeaderMap;
+use serde_json::{json, Value};
+use rocksdb::{DB, IteratorMode, WriteBatch};
+use std::{env, net::SocketAddr, sync::Arc};
+use tracing::{info, error};
+use tracing_subscriber::{layer::SubscriberExt, util::SubscriberInitExt};
+
+mod ip_api;
+mod types;
+
+#[tokio::main]
+async fn main() {
+    tracing_subscriber::registry()
+        .with(tracing_subscriber::EnvFilter::new(
+            std::env::var("RUST_LOG").unwrap_or_else(|_| "info".into()),
+        ))
+        .with(tracing_subscriber::fmt::layer())
+        .init();
+
+    let db = Arc::new(DB::open_default("tracker_db").unwrap());
+    let token = env::var("IPINFO_TOKEN").expect("IPINFO_TOKEN is not set");
+
+    let app = Router::new()
+        .route("/register", post(registration_handler))
+        .route("/http_sd", get(http_sd_handler))
+        .layer(Extension(db))
+        .layer(Extension(token));
+
+    let addr = SocketAddr::from(([0, 0, 0, 0], 4000));
+    println!("Ursa-tracker listening on {}", addr);
+
+    axum::Server::bind(&addr)
+        .serve(app.into_make_service_with_connect_info::<SocketAddr>())
+        .await
+        .unwrap();
+}
+
+/// Track a new peer registration in the database
+async fn registration_handler(
+    headers: HeaderMap,
+    ConnectInfo(req_addr): ConnectInfo<SocketAddr>,
+    db: Extension<Arc<DB>>,
+    token: Extension<String>,
+    Json(registration): Json<TrackerRegistration>,
+) -> (StatusCode, Json<Value>) {
+    let id = registration.id;
+    info!("Received registration for: {}", id);
+
+    // todo: registration verification
```

Comment:
in case the todo is forgotten.

Reply:
Not sure what it'd entail/not required at this time. cc @b0xtch but basically once we have consensus we can switch to that

Reply:
This would be decent verification. Basically send a request to the provided address say for metrics and make sure it responds with 200?

Currently in the source there's no requirement for dns/tls to be setup. Should this be done separately or is that in scope for this pr?

Reply:
Will take care of this with another PR, specific to making changes to enforce DNS/TLS usage by node providers. Exact requirements can be up for discussion